server/src/main/java/org/elasticsearch/common/xcontent/ChunkedToXContentHelper.java

```diff
@@ -119,19 +110,32 @@ public static Iterator<ToXContent> array(String name, Iterator<? extends ToXCont
      * @return Iterator composing field name and value serialization
      */
     public static Iterator<ToXContent> array(String name, Iterator<? extends ChunkedToXContentObject> contents, ToXContent.Params params) {
-        return Iterators.concat(
-            ChunkedToXContentHelper.startArray(name),
-            Iterators.flatMap(contents, c -> c.toXContentChunked(params)),
-            ChunkedToXContentHelper.endArray()
-        );
+        return Iterators.concat(startArray(name), Iterators.flatMap(contents, c -> c.toXContentChunked(params)), endArray());
     }
 
-    public static <T extends ToXContent> Iterator<ToXContent> wrapWithObject(String name, Iterator<T> iterator) {
+    /**
+     * Defines an object named {@code name}, with the contents set by {@code iterator}
+     */
+    public static Iterator<ToXContent> object(String name, Iterator<? extends ToXContent> iterator) {
         return Iterators.concat(startObject(name), iterator, endObject());
     }
 
-    public static <T> Iterator<ToXContent> map(String name, Map<String, T> map, Function<Map.Entry<String, T>, ToXContent> toXContent) {
-        return wrapWithObject(name, Iterators.map(map.entrySet().iterator(), toXContent));
+    /**
+     * Defines an object named {@code name}, with the contents set by calling {@code toXContent} on each entry in {@code map}
+     */
+    public static <T> Iterator<ToXContent> object(String name, Map<String, T> map, Function<Map.Entry<String, T>, ToXContent> toXContent) {
+        return object(name, Iterators.map(map.entrySet().iterator(), toXContent));
+    }
+
+    /**
+     * Creates an Iterator of a single ToXContent object that serializes the given object as a single chunk. Just wraps {@link
+     * Iterators#single}, but still useful because it avoids any type ambiguity.
+     *
+     * @param item Item to wrap
+     * @return Singleton iterator for the given item.
+     */
+    public static Iterator<ToXContent> chunk(ToXContent item) {
+        return Iterators.single(item);
```

Comment:
Nit: should you remove singleChunck? It looks like a leftover

Reply:
It's currently used by Serverless, tidying up Serverless uses I'll do separately.